### PR TITLE
update log creation to work with api v35+, update debug user id after…

### PR DESCRIPTION
**Pull request by ralphcallaway**
… updating creds

fixes https://github.com/joeferraro/MavensMate/issues/602, fixes https://github.com/joeferraro/MavensMate-app/issues/66, and fixes https://github.com/joeferraro/MavensMate-app/issues/105

### CODE REVIEW

**Comment by ralphcallaway** [comment]
@joeferraro got another ones that's ready to merge, in terms of testing
- tried start log with v35 after cleaning out traceflag and debuglevel records
- tried start log with v35 with existing traceflag and debuglevel records
- tried starting logs with v34 with clean/existing traceflag and debuglevel records
- switch credentials for project and start log

not 100% i'm interpretting it correctly, but looks like circle ci timed out one test, but otherwise is passing

let me know if you submit pr #605 and I can rebase this one on top of master

File: lib/mavensmate/sfdc-client.js [hunk 1]
```diff
@@ -903,23 +903,69 @@ SalesforceClient.prototype.startLogging = function(debugSettings, expiration) {
...
+              return self.conn.tooling.query('SELECT Id FROM DebugLevel WHERE DeveloperName = \'MM_DEBUG\'');
+            }
+          })
+          .then(function(debugLevelResult) {
+            logger.debug('debugLevelResult: ' + JSON.stringify(debugLevelResult));
+            var debugLevel = {
+              Workflow: debugLevels.Workflow || 'DEBUG',
+              Callout: debugLevels.Callout || 'DEBUG',
+              System: debugLevels.System || 'DEBUG',
+              Database: debugLevels.Database || 'DEBUG',
```
Comment (joeferraro) [inline lib/mavensmate/sfdc-client.js]: @ralphcallaway FYI, `DEBUG` is actually not a valid value for every one of these levels. Can you verify the valid debug levels (I think you can find them in the standard) debug log creation UI) and update accordingly?

Comment (ralphcallaway) [inline lib/mavensmate/sfdc-client.js]: @joeferraro gotcha, that was based on the existing version, would be good to tackle this at the same time (surprisingly doesn't seem like salesforce complains if we pass in the bad values)

Comment (ralphcallaway) [inline lib/mavensmate/sfdc-client.js]: @joeferraro looks like `DEBUG` is valid for every log category, see [tooling api debug level docs|https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_debuglevel.htm] 

this also matches what's in ui/templates/snippets/tooling_log_levels

fyi, these defaults weren't anything i changed, the big change is that for api version >=35 we need to create a `DebugLevel` record, and then link it to the `TraceFlag` record (as well as add a logType attribute to the `TraceFlag`).

anything else you'd want to see before merging this?

**Comment by ralphcallaway** [comment]
@joeferraro good to merge?

**Pull request closed without merging**